Xamarin.Forms.Core/NavigationPage.cs

```diff
@@ -297,6 +306,9 @@ protected override bool OnBackButtonPressed()
 
 		void InsertPageBefore(Page page, Page before)
 		{
+			if (page == null || before == null)
+				throw new ArgumentException($"Both {nameof(page)} and {nameof(before)} must be initialized.");
```

Comment:
http://stackoverflow.com/a/105021/812582. See Frank's comments. Perhaps it doesn't make sense to unit test all non-public APIs, but `InsertPageBefore` and `RemovePage` are very important to managing the navigation stack. In this particular case, I'd say both the tests and the null checks should be okay.